### PR TITLE
[RFC] process_buffer: call init_buffer with a:force

### PR DESCRIPTION
**Pull request by blueyed**
This is required to re-check if a buffer is tracked now, e.g. after
vim-fugitive's `:Gwrite`.

Test case:

1. have an untracked file, open it
2. gitgutter will not process it, setting "path" to -2
3. call `:Gwrite` to add the file to the index
4. change the buffer
5. signs are not being placed
6. call `:Gitgutter` manually
7. although it calls `gitgutter#process_buffer` with `a:force = 1`, it
   will still not display signs, since `gitgutter#diff#run_diff` throws
   "gitgutter not tracked".

A workaround is to call `gitgutter#init_buffer` yourself manually.

Additionally/instead vim-gitgutter could also call `init_buffer` on
`BufWritePost`, which gets triggered by vim-fugitive, but also on every
other write of course.

### CODE REVIEW

**Comment by airblade** [comment]
The plugin does show signs when a file is added to the index outside vim, i.e. when step 3 above is: go to a terminal, add the file to the index, and go back to vim.  This is because of:

```viml
  autocmd FocusGained,ShellCmdPost * call gitgutter#all(1)
```

And `gitgutter#all(1)` calls `gitgutter#init_buffer()` along the way.

I want vim-gitgutter to update after `:Gwrite` (see #502) but ideally not by listening to `BufWritePost` (because whether or not the buffer is saved is irrelevant to gitgutter).

Anyway, I'll look at your patch.

File: autoload/gitgutter.vim [hunk 1]
```diff
@@ -38,6 +38,10 @@ function! gitgutter#process_buffer(bufnr, force) abort
 
   if gitgutter#utility#is_active(a:bufnr)
     if a:force || s:has_fresh_changes(a:bufnr)
+      if a:force
+        let p = gitgutter#utility#repo_path(a:bufnr, 0)
```
Comment (airblade) [inline autoload/gitgutter.vim]: Why is this line necessary?

Comment (blueyed) [reply autoload/gitgutter.vim]: See the test case above..

I might have mixed up something here though with another PR (e.g. the return value), not sure currently.

**Comment by airblade** [comment]
I think this has been made redundant by #593.
